dockerfile/parser/parser.go

```diff
@@ -331,6 +403,26 @@ func Parse(rwc io.Reader) (*Result, error) {
 	}, nil
 }
 
+func heredocsFromLine(line string) ([]buildkitparser.Heredoc, error) {
+	shlex := buildkitshell.NewLex('\\')
+	shlex.RawQuotes = true
+	shlex.RawEscapes = true
+	shlex.SkipUnsetEnv = true
+	words, _ := shlex.ProcessWords(line, []string{})
```

Comment:
It has ability to preserve RawQuotes and Escapes as it is which i think is missing in imagebuilder's process word's. I think imagebuilder's ProcessWords can be extended to match these features but it will be easier in a new PR.